src/main.rs

```diff
@@ -35,6 +35,11 @@ use structopt::StructOpt;
 
 /// Binary entrypoint, for all CLI subcommands.
 fn main() {
+    ctrlc::set_handler(move || {
```

Comment:
This seems to be placed in the wrong place. It should belong somewhere under `run_agent` I guess?

Reply:
ah, that makes sense, since this is specifically dealing with users attempting to stop the daemon.
I put it in main originally because I just thought we wanted to set up a handler as early as possible.